servers/src/main/java/tachyon/master/next/MasterInfo.java

```diff
@@ -0,0 +1,294 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.next;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.conf.TachyonConf;
+import tachyon.master.next.block.BlockMaster;
+import tachyon.master.next.filesystem.FileSystemMaster;
+import tachyon.master.next.filesystem.meta.InodeFile;
+import tachyon.thrift.BlockInfo;
+import tachyon.thrift.DependencyDoesNotExistException;
+import tachyon.thrift.DependencyInfo;
+import tachyon.thrift.FileBlockInfo;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.FileInfo;
+import tachyon.thrift.InvalidPathException;
+import tachyon.thrift.WorkerInfo;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.underfs.UnderFileSystem.SpaceType;
+
+/**
+ * A wrapper class of BlockMaster, FileSystemMaster and other master modules to provide a view of
+ * their containing information.
+ */
+public final class MasterInfo {
+  private final BlockMaster mBlockMaster;
+  private final FileSystemMaster mFileSystemMaster;
+  private final InetSocketAddress mMasterAddress;
+  private final long mStartTimeMs;
+  private final TachyonConf mTachyonConf;
+  private final String mUFSDataFolder;
+
+  public MasterInfo(BlockMaster blockMaster, FileSystemMaster fileSystemMaster,
+      InetSocketAddress address, TachyonConf tachyonConf) {
+    mBlockMaster = Preconditions.checkNotNull(blockMaster);
+    mFileSystemMaster = Preconditions.checkNotNull(fileSystemMaster);
+    mMasterAddress = Preconditions.checkNotNull(address);
+    mTachyonConf = Preconditions.checkNotNull(tachyonConf);
+    mUFSDataFolder = mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER, Constants.DEFAULT_DATA_FOLDER);
+    mStartTimeMs = System.currentTimeMillis();
+  }
+
+  /**
+   * Gets the capacity of the whole system.
+   *
+   * @return the system's capacity in bytes.
+   */
+  public long getCapacityBytes() {
+    return mBlockMaster.getCapacityBytes();
+  }
+
+  /**
+   * Gets the list of block info of an InodeFile determined by path.
+   *
+   * @param path path to the file
+   * @return The list of the block info of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<BlockInfo> getBlockInfoList(TachyonURI path)
```

Comment:
Why does this method exist? If block info is required for a file, it should be reading using FileBlockInfo.